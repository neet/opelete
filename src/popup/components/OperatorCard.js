import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

export default class OperatorCard extends React.PureComponent {

  static propTypes = {
    operator: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    isHidden: PropTypes.bool,
  }

  render() {
    const { operator, description, isHidden } = this.props;

    return (
      <div className={`operator-list-item ${ isHidden ? 'operator-list-item--hidden' : ''}`}>
        <Button className='operator-list-item__hide-button' onClick={this.handleToggleVisibility}>
          <i className='fas fa-eye-slash' aria-hidden='true' />
        </Button>

        <div className='operator-list-item__meta'>
          <h3 className='operator-list-item__operator'>
            { operator }
          </h3>

          <p className='operator-list-item__description'>
            { description }
          </p>
        </div>
      </div>
    );
  }

}
