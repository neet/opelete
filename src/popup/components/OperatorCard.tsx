import { faEye, faEyeSlash } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon as Fa } from '@fortawesome/react-fontawesome';
import React from 'react';
import { Operator } from '../../opelete/operators';
import Button from './Button';

interface Props {
  operator: Operator;
  isHidden: boolean;
  onAddToBlacklist: () => void;
  onRemoveFromBlacklist: () => void;
}

export default class OperatorCard extends React.PureComponent<Props> {

  public handleAddToBlacklist = () => {
    this.props.onAddToBlacklist();
  }

  public handleRemoveFromBlacklist = () => {
    this.props.onRemoveFromBlacklist();
  }

  public render () {
    const { operator, isHidden } = this.props;

    return (
      <div className={`operator-list-item ${isHidden ? 'operator-list-item--hidden' : ''}`}>
        <Button className='operator-list-item__hide-button' onClick={isHidden ? this.handleRemoveFromBlacklist : this.handleAddToBlacklist}>
          <Fa icon={isHidden ? faEyeSlash : faEye} />
        </Button>

        <div className='operator-list-item__meta'>
          <h3 className='operator-list-item__operator'>
            {operator.operator}
          </h3>

          <p className='operator-list-item__description'>
            {operator.description}
          </p>
        </div>
      </div>
    );
  }

}
