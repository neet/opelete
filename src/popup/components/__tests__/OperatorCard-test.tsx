import * as React from 'react';
import OperatorCard from '../OperatorCard';
import renderer from 'react-test-renderer';

describe('OperatorCard', () => {
  it('renders a checkbox', () => {
    const tree = renderer.create(
      <OperatorCard
        operator={{
          id: 'site',
          operator: 'site:',
          description: 'Search for a specific site',
        }}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
