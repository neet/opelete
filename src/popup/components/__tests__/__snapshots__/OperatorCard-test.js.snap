// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperatorCard renders a checkbox 1`] = `
<div
  className="operator-list-item "
>
  <button
    aria-label={undefined}
    className="button operator-list-item__hide-button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fas fa-eye-slash"
    />
  </button>
  <div
    className="operator-list-item__meta"
  >
    <h3
      className="operator-list-item__operator"
    >
      site:
    </h3>
    <p
      className="operator-list-item__description"
    >
      Search for a specific site
    </p>
  </div>
</div>
`;
